src/Spectre.libClassifier/DatasetFilter.h

```diff
@@ -0,0 +1,41 @@
+/*
+* DatasetFilter.h
+* ENTER CLASS DESCRIPTION HERE.
+*
+Copyright 2018 Spectre Team
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#pragma once
+#include <vector>
+#include "Spectre.libFunctional/Filter.h"
+#include "OpenCvDataset.h"
+
+namespace spectre::supervised
+{
+inline OpenCvDataset getFilteredOpenCvDataset(const OpenCvDataset* dataset, std::vector<bool>& filterPattern)
```

Comment:
You're passing `const OpenCvDataset*`. What if `dataset == nullptr`? Either use `const OpenCvDataset&` or check for `nullptr`. The first would decrease branching.

Reply:
you're right, done